netmaster/master/api.go

```diff
@@ -88,6 +92,10 @@ func AllocAddressHandler(w http.ResponseWriter, r *http.Request, vars map[string
 
 	log.Infof("Received AddressAllocRequest: %+v", allocReq)
 
+	// Take a global lock for address allocation
+	addrMutex.Lock()
+	defer addrMutex.Unlock()
```

Comment:
No, its for multiple REST calls coming into same net master at the same time. This was seen while testing "docker-compose scale" command. docker-compose multiple containers in parallel and some of them end up getting duplicate addresses.